### ISSUE TITLE
Feature request: categories/sections

### ISSUE THREAD

**Issue by leviska** [casey/just#1344]
I think it would be good to have some sections in `just -l`
For example, output of my file is
```
Available recipes:
    default                     # list all recipes
    build cmd                   # build go binary locally
    docker-build cmd            # build single cmd docker image and tag is as cmd
    docker-run cmd *flags       # run image with tag based on cmd
    docker-build-run cmd *flags # docker-build and then docker-run
    compose-build cmd           # build and push a compose file
    deploy cmd context_name="master" # deploy compose file
    compose-build-deploy cmd context_name="master" # compose-build and then deploy
```
And I would like to be able to make something like
```
Available recipes:
    just:
        default                     # list all recipes
    build:
        build cmd                   # build go binary locally
    docker:
        docker-build cmd            # build single cmd docker image and tag is as cmd
        docker-run cmd *flags       # run image with tag based on cmd
        docker-build-run cmd *flags # docker-build and then docker-run
    deploy:
        compose-build cmd           # build and push a compose file
        deploy cmd context_name="master" # deploy compose file
        compose-build-deploy cmd context_name="master" # compose-build and then deploy
```

In file it could look something like

```make
# just-category: build
...build recipes
# just-category: docker
...docker recipes
```

or 

```make
# build go binary locally #category:build
build cmd:
...
# build single cmd docker image and tag is as cmd #category:docker
docker cmd:
...
```

Or whatever you like :) 

Or may be even make nested recipes, so you could call something like
```bash
just build # same stuff
just docker build # instead of `just docker-build`
just docker run # instead of `just docker-run`
just deploy # same
just deploy build # instead of `just compose-build`
```

**Comment by runeimp** [comment]
I like the idea but it smacks of namespacing and once Just adds module support the category/section stuff would likely further be altered in the list output. May not be a big deal, no idea, but definitely seems like something to consider in that context. Don't mind me. Just thinking out-loud.  👼

**Comment by casey** [comment]
I'm inclined to agree with @runeimp, although since module support is far away, I wouldn't mind low-impact things that made large justfiles easier to navigate.

How about some kind of section marker that you can put in a justfile that if used gets added to `--list` output and disables alphabetic sorting of `--list`.

So you can do:

```
!! A

foo:

!! B

bar:

!! C

baz:
```

And then `just --list` looks like:

```
$ just --list
Available recipes:
  # A
  foo
  # B
  bar
  # C
  baz
```

No idea what the notation would look like, and indeed, finding a notation that's unambiguously good would be pretty important for making me want to add this.

**Comment by dheeraj-squadcast** [comment]
+1 to this. I need a way to separate tasks in the task list using a section title and a newline. I've hacked around it by using ascii art (!) like so:

```
_default:
  @just --list --unsorted

@x-----Initial-Setup-----x:

... most important task

@x-----Everyday-Tasks----x:

... some tasks

@x-----Miscellaneous-----x:

...more tasks
```

It looks like this:

```
$ just
Available recipes:
    x-----Initial-Setup-----x
    db-setup                  # Setup the dev db from scratch
    x-----Everyday-Tasks----x
    make-migration name       # Create a new db migration under db/migrations/
    db-migrate                # Run the latest migrations
    db-migrate-status         # Show the migration status
    x-----Miscellaneous-----x
    deps                      # Install all binary dependencies
    db-create                 # Create the local dev db
    db-drop                   # Drop the local dev db
    db-schema-dump            # Dump the schema to db/schema.sql
```

**Comment by runeimp** [comment]
My one concern with @casey's suggestion be that the sections be alphabetically sorted and the recipes be  sorted within their sections.  😇

**Comment by dheeraj-squadcast** [comment]
In that case, we might need a counterpart of --unsorted for sections as well.

**Comment by casey** [comment]
Implemented in #1842!
